toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -20,29 +26,62 @@ func extractPartitions(imageLoopDevice string, outDir string, basename string, p
 		return err
 	}
 
+	// Define magicNumber and frameSize for skippable frames
+	magicNumber := uint32(0x184D2A50)
+	frameSize := uint32(16)
+
+	// Create skippable frame metadata defined as a random 128-Bit number
+	metadata, err := createSkippableFrameMetadata()
+	if err != nil {
+		return err
+	}
+
+	// Encode metadata into a payload defined as a uint32 array
+	payload := Encode128BitLittleEndian(metadata)
+
 	for partitionNum := 0; partitionNum < len(diskPartitions); partitionNum++ {
 		if diskPartitions[partitionNum].Type == "part" {
 			rawFilename := basename + "_" + strconv.Itoa(partitionNum) + ".raw"
 			partitionLoopDevice := diskPartitions[partitionNum].Path
 
-			partitionFilepath, err := copyBlockDeviceToFile(outDir, partitionLoopDevice, rawFilename)
+			partitionRawFilepath, err := copyBlockDeviceToFile(outDir, partitionLoopDevice, rawFilename)
 			if err != nil {
 				return err
 			}
 
 			switch partitionFormat {
 			case "raw":
 				// Do nothing for "raw" case.
+				logger.Log.Infof("Partition file created: %s", partitionRawFilepath)
 			case "raw-zst":
-				partitionFilepath, err = compressWithZstd(partitionFilepath)
```

Comment:
To avoid renaming on line 47 and moving/copying logger.Log.Infof lines, have you tried:

```
case "raw-zst":
  partitionRawFilePath := partitionFilePath
  partitionFilepath, err = compressWithZstd(partitionFilepath)
  ...
```

Reply:
Good point, I will add this change in

Reply:
Added this in, resolving comment.